### ISSUE TITLE
Update the WithAuth Example to Use a Mock Client

### ISSUE THREAD

**Issue by prestonvasquez** [alpstable/gidari#395]
There is no reason to actually make an HTTP request in ExampleWithAuth.

**Comment by daredevil3435** [comment]
Can you explain issue more? I would like to contribute

**Comment by prestonvasquez** [comment]
@daredevil3435 Sure, we should probably mock http requests for the testable examples. This would look something like the following:

```go
type MockClient {}

func (mclient *MockClient) Do(req *http.Request) (*http.Reqponse, error) { 
         return &http.Response{Request: req}, nil
}

func ExampleWithAuth() {
        // ...
    	svc, err := gidari.NewService(ctx)
	if err != nil {
		log.Fatalf("failed to create service: %v", err)
	}
        
        svc.Client(&MockClient{})
        // ...
}
```

**Comment by daredevil3435** [comment]
Yes I would like to work, no need to assign me. Exactly where do I need to add this?

**Comment by prestonvasquez** [comment]
@daredevil3435 This is an update to gidari_example_test.go in the "ExampleWithAuth" function.

**Comment by daredevil3435** [comment]
so I've to replace 

func ExampleWithAuth() {
	ctx := context.TODO()

	const api = "https://api-public.sandbox.exchange.coinbase.com"

	key := os.Getenv("COINBASE_API_KEY")
	secret := os.Getenv("COINBASE_API_SECRET")
	passphrase := os.Getenv("COINBASE_API_PASSPHRASE")

	// First we create a service that can be used to make bulk HTTP
	// requests to the API.
	svc, err := gidari.NewService(ctx)
	if err != nil {
		log.Fatalf("failed to create service: %v", err)
	}

	// Set a round tripper that will sign the requests.
	roundTripper := auth.NewCoinbaseRoundTrip(key, secret, passphrase)
	withAuth := gidari.WithAuth(roundTripper)

	// Create some requests and add them to the service.
	accounts, _ := http.NewRequestWithContext(ctx, http.MethodGet, api+"/accounts", nil)
	currencies, _ := http.NewRequestWithContext(ctx, http.MethodGet, api+"/currencies", nil)

	// Wrap the HTTP Requests in the gidalri.HTTPRequest type.
	accountsW := gidari.NewHTTPRequest(accounts, withAuth)
	currenciesW := gidari.NewHTTPRequest(currencies, withAuth)

	// Add the wrapped HTTP requests to the HTTP Service.
	svc.HTTP.Requests(accountsW, currenciesW)

	// Get the status code for the responses.
	for svc.HTTP.Iterator.Next(ctx) {
		current := svc.HTTP.Iterator.Current

		rsp := current.Response
		if rsp == nil {
			break
		}

		fmt.Println("status code:", rsp.Request.URL.Path, rsp.StatusCode)
	}
}

with the above code?

**Comment by daredevil3435** [comment]
I'm kind of newbie in go so asking you in so much details. I'm sorry for that in advance

**Comment by prestonvasquez** [comment]
@daredevil3435 No worries. The problem here is that we are actually making a request to 

```
const api = "https://api-public.sandbox.exchange.coinbase.com"
```

And we don't want to do that. Instead, we should create a mock client, like in [this](https://github.com/alpstable/gidari/issues/395#issuecomment-1441129664) example, and set that on the service.

**Comment by daredevil3435** [comment]
> @daredevil3435 No worries. The problem here is that we are actually making a request to 
> 
> 
> 
> ```
> 
> const api = "https://api-public.sandbox.exchange.coinbase.com"
> 
> ```
> 
> 
> 
> And we don't want to do that. Instead, we should create a mock client, like in [this](https://github.com/alpstable/gidari/issues/395#issuecomment-1441129664) example, and set that on the service. 
> 
> 
> 
> 

I'll try and update you
